### DISCUSSION TITLE
storage-rn/commonjs bug

### DISCUSSION THREAD

**Discussion by kamilafsar** [urql-graphql/urql#3248] [Q&A]
Hi there,

I'm using `makeAsyncStorage` from the "@urql/storage-rn" package in a React Native app.
Everything works quite nice exept for "offline support". After a reboot of the app it wipes the cache.

After debugging the transpiled JS it turns out that `makeAsyncStorage` expects `setItem`/`getItem` like methods directly under the required `AsyncStorage` variable. But because it's transpiled into CommonJS the AsyncStorage is hidden in a `default` property.

So changing:
`AsyncStorage.setItem(...)`
into:
`AsyncStorage.default.setItem(...)`
works.

Not entirely sure if this is a bug or did I somehow import the wrong transpiled version?
There's also a `.mjs` version in the dist directory, but I'm not entirely sure how I could use that particular version.

**Comment by sighmun** [comment]
I just encountered this same problem and was about to file a bug, but then I saw this discussion. I would've expected the `require` in `urql-storage-rn.js` to have transpiled out to:
 `var AsyncStorage = require('@react-native-async-storage/async-storage').default;`
instead of:
`var AsyncStorage = require('@react-native-async-storage/async-storage');`

For now, I've updated my `babel.config.js` to import the `.mjs` version, which works fine.

`babel.config.js`:

```
module.exports = function (api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
    plugins: [
      [
        'module-resolver',
        {
          alias: {
            '@urql/storage-rn': './node_modules/@urql/storage-rn/dist/urql-storage-rn.mjs',
          },
        },
      ],
    ],
  };
};

```

**Reply by kamilafsar** [reply]
Ah, that's actually an excellent workaround! Thanks for your reply.
Well I guess it is a bug then.

**Comment by kamilafsar** [comment]
@strewhella 
Seems like there's a fix on its way: https://github.com/urql-graphql/urql/pull/3252
